src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -446,17 +446,10 @@ public void StopStake()
                 return;
             }
 
-            this.stakeCancellationTokenSource.Cancel();
-            this.logger.LogTrace("Waiting for staking loop to cancel...");
-            this.stakingLoop?.RunningTask?.GetAwaiter().OnCompleted(() =>
-            {
-                this.logger.LogTrace("Staking loop cancelled.");                              
-            });
+            this.stakeCancellationTokenSource.Cancel();  
```

Comment:
although if it is possible on line 452 that sCTS is null, it is also possible here
so is it possible or not?